src/hypothesis/core.py

```diff
@@ -880,6 +884,28 @@ def __flaky(self, message):
             report('Flaky example! ' + message)
 
 
+@contextlib.contextmanager
+def fake_subTest(self, msg=None, **__):
+    """Monkeypatch for `unittest.TestCase.subTest` during `@given`.
+
+    If we don't patch this out, each failing example is reported as a
+    seperate failing test by the unittest test runner, which is
+    obviously incorrect. We therefore replace it for the duration with
+    this version.
+
+    """
+    note_deprecation(
+        'Using unittest.TestCase.subTest with Hypothesis is deprecated.  '
+        'subTest per-example reporting interacts badly with Hypothesis '
+        'trying hundreds of examples, so we disable it for the duration of '
+        'any test that uses `@given`.  You should therefore remove it from '
+        'this test, to avoid misleading people reading the code - and '
+        'because this will be an error in a future version of Hypothesis!  '
```

Comment:
Continuing from https://github.com/HypothesisWorks/hypothesis-python/pull/1072#discussion_r171136504:

I think it's a pretty interesting question whether this should become an error. The tradeoffs seem to be:

- an error is better than misleading/surprising behavior
- but it's nice to just be able to swap in hypothesis strategies and have everything still work (and it isn't *that* bad to not see all of the reasons a failed test failed).

A warning that's *not* a deprecation warning might be a good idea.

Since a deprecation warning doesn't force us to remove anything later, I'm not too inclined to object or ask for more discussion now, but when it actually comes time to remove it I expect I might make the case for turning it into a (non-deprecation) warning instead.

Reply:
I'll try to get around to doing a full review of this at some point, but I *really* don't think we should deprecate subtest! I don't think subtests with Hypothesis are an unreasonable thing to want and more than pytest fixtures with Hypothesis are, and I don't think there's any significant technical challenge to us supporting them in future (beyond just "It's a bunch of annoying work and we don't care to do it").

I'd be OK with the suggestion of a warning that is not a deprecation. `PatchesWelcomeWarning` or something. 😉 (please don't call it that)

Reply:
I'll add `HypothesisWarning(HypothesisException, Warning)` and use that then :smile: